In [35]:
import pandapower as pp

#create empty net
net = pp.create_empty_network()

#create buses
b1 = pp.create_bus(net, vn_kv=110, name="bus 1")
b2 = pp.create_bus(net, vn_kv=20, name="bus 2")
b3 = pp.create_bus(net, vn_kv=20, name="bus 3")
b4 = pp.create_bus(net, vn_kv=20, name="bus 4")
b5 = pp.create_bus(net, vn_kv=20, name="bus 5")
b6 = pp.create_bus(net, vn_kv=20, name="bus 6")

b41 = pp.create_bus(net, vn_kv=0.4, name="bus 41")
b42 = pp.create_bus(net, vn_kv=0.4, name="bus 42")
b43 = pp.create_bus(net, vn_kv=0.4, name="bus 43")
b44 = pp.create_bus(net, vn_kv=0.4, name="bus 44")
b45 = pp.create_bus(net, vn_kv=0.4, name="bus 45")
b46 = pp.create_bus(net, vn_kv=0.4, name="bus 46")
b47 = pp.create_bus(net, vn_kv=0.4, name="bus 47")
b48 = pp.create_bus(net, vn_kv=0.4, name="bus 48")
b49 = pp.create_bus(net, vn_kv=0.4, name="bus 49")

b51 = pp.create_bus(net, vn_kv=0.4, name="bus 51")
b52 = pp.create_bus(net, vn_kv=0.4, name="bus 52")
b53 = pp.create_bus(net, vn_kv=0.4, name="bus 53")
b54 = pp.create_bus(net, vn_kv=0.4, name="bus 54")
b55 = pp.create_bus(net, vn_kv=0.4, name="bus 55")
b56 = pp.create_bus(net, vn_kv=0.4, name="bus 56")
b57 = pp.create_bus(net, vn_kv=0.4, name="bus 57")
b58 = pp.create_bus(net, vn_kv=0.4, name="bus 58")
b59 = pp.create_bus(net, vn_kv=0.4, name="bus 59")

b61 = pp.create_bus(net, vn_kv=0.4, name="bus 61")
b62 = pp.create_bus(net, vn_kv=0.4, name="bus 62")
b63 = pp.create_bus(net, vn_kv=0.4, name="bus 63")
b64 = pp.create_bus(net, vn_kv=0.4, name="bus 64")
b65 = pp.create_bus(net, vn_kv=0.4, name="bus 65")
b66 = pp.create_bus(net, vn_kv=0.4, name="bus 66")
b67 = pp.create_bus(net, vn_kv=0.4, name="bus 67")
b68 = pp.create_bus(net, vn_kv=0.4, name="bus 68")
b69 = pp.create_bus(net, vn_kv=0.4, name="bus 69")



In [36]:
#create external grid
pp.create_ext_grid(net, b1, s_sc_max_mva=100., s_sc_min_mva=80., rx_max=0.35, rx_min=0.20, name="External Grid")

0

In [37]:
#create transformers
trafo1 = pp.create_transformer(net, b1, b2, std_type="25 MVA 110/20 kV", name="trafo1")
trafo2 = pp.create_transformer(net, b4, b41, std_type="0.4 MVA 20/0.4 kV", name="trafo2")
trafo3 = pp.create_transformer(net, b5, b51, std_type="0.4 MVA 20/0.4 kV", name="trafo3")
trafo4 = pp.create_transformer(net, b6, b61, std_type="0.4 MVA 20/0.4 kV", name="trafo4")



In [38]:
#create line
line1 = pp.create_line(net, b2, b3, std_type="NA2XS2Y 1x95 RM/25 12/20 kV", length_km=5., name="line1")
line2 = pp.create_line(net, b2, b4, std_type="NA2XS2Y 1x95 RM/25 12/20 kV", length_km=6., name="line2")
line3 = pp.create_line(net, b2, b5, std_type="NA2XS2Y 1x95 RM/25 12/20 kV", length_km=7., name="line3")
line4 = pp.create_line(net, b2, b6, std_type="NA2XS2Y 1x95 RM/25 12/20 kV", length_km=6., name="line4")

line5 = pp.create_line(net, b41, b42, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line5")
line6 = pp.create_line(net, b42, b43, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line6")
line7 = pp.create_line(net, b43, b44, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line7")
line8 = pp.create_line(net, b44, b45, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line8")
line9 = pp.create_line(net, b45, b46, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line9")
line10 = pp.create_line(net, b46, b47, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line10")
line11 = pp.create_line(net, b47, b48, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line11")
line12 = pp.create_line(net, b48, b49, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line12")

line13 = pp.create_line(net, b51, b52, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line13")
line14 = pp.create_line(net, b52, b53, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line14")
line15 = pp.create_line(net, b53, b54, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line15")
line16 = pp.create_line(net, b54, b55, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line16")
line17 = pp.create_line(net, b55, b56, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line17")
line18 = pp.create_line(net, b56, b57, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line18")
line19 = pp.create_line(net, b57, b58, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line19")
line20 = pp.create_line(net, b58, b59, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line20")

line21 = pp.create_line(net, b61, b62, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line21")
line22 = pp.create_line(net, b62, b63, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line22")
line23 = pp.create_line(net, b63, b64, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line23")
line24 = pp.create_line(net, b64, b65, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line24")
line25 = pp.create_line(net, b65, b66, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line25")
line26 = pp.create_line(net, b66, b67, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line26")
line27 = pp.create_line(net, b67, b68, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line27")
line28 = pp.create_line(net, b68, b69, std_type="48-AL1/8-ST1A 0.4", length_km=0.1, name="line28")

In [39]:
#create motor
motor = pp.create_sgen(net, b3, p_mw=3, sn_mva=3.162, k=1.2, q_mvar=1, rx=7, name="motor")

In [40]:
#create load
load1 = pp.create_load(net, b42, p_mw=0.0022, q_mvar=0.00072, name="load1" )
load2 = pp.create_load(net, b43, p_mw=0.0022, q_mvar=0.00072, name="load2" )
load3 = pp.create_load(net, b44, p_mw=0.0022, q_mvar=0.00072, name="load3" )
load4 = pp.create_load(net, b45, p_mw=0.0022, q_mvar=0.00072, name="load4" )
load5 = pp.create_load(net, b46, p_mw=0.0022, q_mvar=0.00072, name="load5" )
load6 = pp.create_load(net, b47, p_mw=0.0022, q_mvar=0.00072, name="load6" )
load7 = pp.create_load(net, b48, p_mw=0.0022, q_mvar=0.00072, name="load7" )
load8 = pp.create_load(net, b49, p_mw=0.0022, q_mvar=0.00072, name="load8" )

load9 = pp.create_load(net, b52, p_mw=0.0022, q_mvar=0.00072, name="load9" )
load10 = pp.create_load(net, b53, p_mw=0.0022, q_mvar=0.00072, name="load10" )
load11 = pp.create_load(net, b54, p_mw=0.0022, q_mvar=0.00072, name="load11" )
load12 = pp.create_load(net, b55, p_mw=0.0022, q_mvar=0.00072, name="load12" )
load13 = pp.create_load(net, b56, p_mw=0.0022, q_mvar=0.00072, name="load13" )
load14 = pp.create_load(net, b57, p_mw=0.0022, q_mvar=0.00072, name="load14" )
load15 = pp.create_load(net, b58, p_mw=0.0022, q_mvar=0.00072, name="load15" )
load16 = pp.create_load(net, b59, p_mw=0.0022, q_mvar=0.00072, name="load16" )

load17 = pp.create_load(net, b62, p_mw=0.0022, q_mvar=0.00072, name="load17" )
load18 = pp.create_load(net, b63, p_mw=0.0022, q_mvar=0.00072, name="load18" )
load19 = pp.create_load(net, b64, p_mw=0.0022, q_mvar=0.00072, name="load19" )
load20 = pp.create_load(net, b65, p_mw=0.0022, q_mvar=0.00072, name="load20" )
load21 = pp.create_load(net, b66, p_mw=0.0022, q_mvar=0.00072, name="load21" )
load22 = pp.create_load(net, b67, p_mw=0.0022, q_mvar=0.00072, name="load22" )
load23 = pp.create_load(net, b68, p_mw=0.0022, q_mvar=0.00072, name="load23" )
load24 = pp.create_load(net, b69, p_mw=0.0022, q_mvar=0.00072, name="load24" )



# power flow modeling and Analysis
# 2.2 Tasks
1.
 A. The type of the 20kv lines used is NA2XS2Y 1x95 RM/25
  Information from the  manufacturer are
  r_ohm_per_km =0.313
  x_ohm_per_km =0.132
  c_nf_per_km  =216.0
  max_i_ka     =0.27
  type         =cs
  q_mm2        =150
  alpha        =0.00403
 yes, This is typical for MV systems as according to  details given by the manufacturer
 
  
 B. The type of the 400v lines used are 48-AL1/8-ST1A 
   Information from the manufacturers are   
   r_ohm_per_km =0.5939
   x_ohm_per_km =0.3
   c_nf_per_km  =12.2
   max_i_ka     =0.21
   type         =ol
   q_mm2        =48
   alpha        =0.00403
   
   Yes ,Thi is typical for LV systems as according to the details given by the manufacturer
   
   
  C. This is radial system and it is typical for disturbtion system.
     As we can see in the diagram that feedrers are directly connected .

In [41]:
pp.runpp(net)


In [42]:
net.res_bus

vm_pu  va_degree      p_mw    q_mvar
0   1.000000   0.000000  2.887723  1.567084
1   1.008110   0.774626  0.000000  0.000000
2   1.021332   0.816950 -3.000000 -1.000000
3   1.008170   0.752150  0.000000  0.000000
4   1.008211   0.744143  0.000000  0.000000
5   1.008170   0.752150  0.000000  0.000000
6   1.006640   0.610264  0.000000  0.000000
7   0.998882   0.544044  0.002200  0.000720
8   0.992078   0.485251  0.002200  0.000720
9   0.986234   0.434212  0.002200  0.000720
10  0.981355   0.391214  0.002200  0.000720
11  0.977446   0.356506  0.002200  0.000720
12  0.974512   0.330293  0.002200  0.000720
13  0.972554   0.312730  0.002200  0.000720
14  0.971574   0.303922  0.002200  0.000720
15  1.006682   0.602268  0.000000  0.000000
16  0.998924   0.536054  0.002200  0.000720
17  0.992120   0.477267  0.002200  0.000720
18  0.986276   0.426231  0.002200  0.000720
19  0.981398   0.383237  0.002200  0.000720
20  0.977489   0.348533  0.002200  0.000720
21  0.974555   0.322322  0.002200  0.000720
22  0.972597   0.304760  0.002200  0.000720
23  0.971618   0.295953  0.002200  0.000720
24  1.006640   0.610264  0.000000  0.000000
25  0.998882   0.544044  0.002200  0.000720
26  0.992078   0.485251  0.002200  0.000720
27  0.986234   0.434212  0.002200  0.000720
28  0.981355   0.391214  0.002200  0.000720
29  0.977446   0.356506  0.002200  0.000720
30  0.974512   0.330293  0.002200  0.000720
31  0.972554   0.312730  0.002200  0.000720
32  0.971574   0.303922  0.002200  0.000720

In [43]:
net.res_trafo

p_hv_mw  q_hv_mvar   p_lv_mw  q_lv_mvar     pl_mw   ql_mvar   i_hv_ka  \
0 -2.887723  -1.567084  2.903616   1.629728  0.015892  0.062644  0.017245   
1  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   
2  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   
3  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   

    i_lv_ka  vm_hv_pu  va_hv_degree  vm_lv_pu  va_lv_degree  loading_percent  
0  0.095347  1.000000      0.000000  1.008110      0.774626        13.211717  
1  0.027226  1.008170      0.752150  1.006640      0.610264         5.039442  
2  0.027225  1.008211      0.744143  1.006682      0.602268         5.039250  
3  0.027226  1.008170      0.752150  1.006640      0.610264         5.039442

In [44]:
net.res_line

p_from_mw  q_from_mvar   p_to_mw  q_to_mvar     pl_mw   ql_mvar  \
0   -2.961942    -1.123698  3.000000   1.000000  0.038058 -0.123698   
1    0.019436    -0.159481 -0.019408  -0.006029  0.000029 -0.165510   
2    0.019454    -0.187069 -0.019408  -0.006029  0.000046 -0.193098   
3    0.019436    -0.159481 -0.019408  -0.006029  0.000029 -0.165510   
4    0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
5    0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
6    0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
7    0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
8    0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
9    0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
10   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
11   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   
12   0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
13   0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
14   0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
15   0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
16   0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
17   0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
18   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
19   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   
20   0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
21   0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
22   0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
23   0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
24   0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
25   0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
26   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
27   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   

    i_from_ka   i_to_ka      i_ka  vm_from_pu  va_from_degree  vm_to_pu  \
0    0.090715  0.089380  0.090715    1.008110        0.774626  1.021332   
1    0.004601  0.000582  0.004601    1.008110        0.774626  1.008170   
2    0.005386  0.000582  0.005386    1.008110        0.774626  1.008211   
3    0.004601  0.000582  0.004601    1.008110        0.774626  1.008170   
4    0.027226  0.027226  0.027226    1.006640        0.610264  0.998882   
5    0.023881  0.023881  0.023881    0.998882        0.544044  0.992078   
6    0.020513  0.020513  0.020513    0.992078        0.485251  0.986234   
7    0.017126  0.017126  0.017126    0.986234        0.434212  0.981355   
8    0.013721  0.013721  0.013721    0.981355        0.391214  0.977446   
9    0.010303  0.010303  0.010303    0.977446        0.356506  0.974512   
10   0.006874  0.006874  0.006874    0.974512        0.330293  0.972554   
11   0.003439  0.003439  0.003439    0.972554        0.312730  0.971574   
12   0.027225  0.027225  0.027225    1.006682        0.602268  0.998924   
13   0.023880  0.023880  0.023880    0.998924        0.536054  0.992120   
14   0.020513  0.020513  0.020513    0.992120        0.477267  0.986276   
15   0.017125  0.017125  0.017125    0.986276        0.426231  0.981398   
16   0.013720  0.013720  0.013720    0.981398        0.383237  0.977489   
17   0.010302  0.010302  0.010302    0.977489        0.348533  0.974555   
18   0.006874  0.006874  0.006874    0.974555        0.322322  0.972597   
19   0.003439  0.003439  0.003439    0.972597        0.304760  0.971618   
20   0.027226  0.027226  0.027226    1.006640        0.610264  0.998882   
21   0.023881  0.023881  0.023881    0.998882        0.544044  0.992078   
22   0.020513  0.020513  0.020513    0.992078        0.485251  0.986234   
23   0.017126  0.017126  0.017126    0.986234        0.434212  0.981355   
24   0.013721  0.013721  0.013721    0.981355        0.391214  0.9774

In [45]:
net.load

name  bus    p_mw   q_mvar  const_z_percent  const_i_percent  sn_mva  \
0    load1    7  0.0022  0.00072              0.0              0.0     NaN   
1    load2    8  0.0022  0.00072              0.0              0.0     NaN   
2    load3    9  0.0022  0.00072              0.0              0.0     NaN   
3    load4   10  0.0022  0.00072              0.0              0.0     NaN   
4    load5   11  0.0022  0.00072              0.0              0.0     NaN   
5    load6   12  0.0022  0.00072              0.0              0.0     NaN   
6    load7   13  0.0022  0.00072              0.0              0.0     NaN   
7    load8   14  0.0022  0.00072              0.0              0.0     NaN   
8    load9   16  0.0022  0.00072              0.0              0.0     NaN   
9   load10   17  0.0022  0.00072              0.0              0.0     NaN   
10  load11   18  0.0022  0.00072              0.0              0.0     NaN   
11  load12   19  0.0022  0.00072              0.0              0.0     NaN   
12  load13   20  0.0022  0.00072              0.0              0.0     NaN   
13  load14   21  0.0022  0.00072              0.0              0.0     NaN   
14  load15   22  0.0022  0.00072              0.0              0.0     NaN   
15  load16   23  0.0022  0.00072              0.0              0.0     NaN   
16  load17   25  0.0022  0.00072              0.0              0.0     NaN   
17  load18   26  0.0022  0.00072              0.0              0.0     NaN   
18  load19   27  0.0022  0.00072              0.0              0.0     NaN   
19  load20   28  0.0022  0.00072              0.0              0.0     NaN   
20  load21   29  0.0022  0.00072              0.0              0.0     NaN   
21  load22   30  0.0022  0.00072              0.0              0.0     NaN   
22  load23   31  0.0022  0.00072              0.0              0.0     NaN   
23  load24   32  0.0022  0.00072              0.0              0.0     NaN   

    scaling  in_service type  
0       1.0        True  wye  
1       1.0        True  wye  
2       1.0        True  wye  
3       1.0        True  wye  
4       1.0        True  wye  
5       1.0        True  wye  
6       1.0        True  wye  
7       1.0        True  wye  
8       1.0        True  wye  
9       1.0        True  wye  
10      1.0        True  wye  
11      1.0        True  wye  
12      1.0        True  wye  
13      1.0        True  wye  
14      1.0        True  wye  
15      1.0        True  wye  
16      1.0        True  wye  
17      1.0        True  wye  
18      1.0        True  wye  
19      1.0        True  wye  
20      1.0        True  wye  
21      1.0        True  wye  
22      1.0        True  wye  
23      1.0        True  wye

2.
A. There are no violations in the volatge,line & transformers.
 
B. Create and display a colormap of the buses showing blue for all voltages below 
0.95, green for voltages between 0.95 and 1.05, and red for all voltages above 
1.05 pu.


In [ ]:
import pandapower.plotting as plot
pp.runpp(net)
cmap_list = [((0, 0.95), "blue"), ((0.95, 1.05), "green"), ((1.05, 1.1), "red")]
cmap, norm = plot.cmap_discrete(cmap_list)
bc = plot.create_bus_collection(net, net.bus.index, size=0.1, zorder=2, cmap=cmap, norm=norm)
lc = create_line_collection(net)
plot.draw_collections([lc, bc])

 3.
 A. Every household increased to 4Kw at 0.95 power factor
 

In [ ]:
#create load
load1 = pp.create_load(net, b42, p_mw=0.0018, q_mvar=0.00058, name="load1" )
load2 = pp.create_load(net, b43, p_mw=0.0018, q_mvar=0.00058, name="load2" )
load3 = pp.create_load(net, b44, p_mw=0.0018, q_mvar=0.00058, name="load3" )
load4 = pp.create_load(net, b45, p_mw=0.0018, q_mvar=0.00058, name="load4" )
load5 = pp.create_load(net, b46, p_mw=0.0018, q_mvar=0.00058, name="load5" )
load6 = pp.create_load(net, b47, p_mw=0.0018, q_mvar=0.00058, name="load6" )
load7 = pp.create_load(net, b48, p_mw=0.0018, q_mvar=0.00058, name="load7" )
load8 = pp.create_load(net, b49, p_mw=0.0018, q_mvar=0.00058, name="load8" )

load9 = pp.create_load(net, b52, p_mw=0.0018, q_mvar=0.00058, name="load9" )
load10 = pp.create_load(net, b53, p_mw=0.0018, q_mvar=0.00058, name="load10" )
load11 = pp.create_load(net, b54, p_mw=0.0018, q_mvar=0.00058, name="load11" )
load12 = pp.create_load(net, b55, p_mw=0.0018, q_mvar=0.00058, name="load12" )
load13 = pp.create_load(net, b56, p_mw=0.0018, q_mvar=0.00058, name="load13" )
load14 = pp.create_load(net, b57, p_mw=0.0018, q_mvar=0.00058, name="load14" )
load15 = pp.create_load(net, b58, p_mw=0.0018, q_mvar=0.00058, name="load15" )
load16 = pp.create_load(net, b59, p_mw=0.0018, q_mvar=0.00058, name="load16" )

load17 = pp.create_load(net, b62, p_mw=0.0018, q_mvar=0.00058, name="load17" )
load18 = pp.create_load(net, b63, p_mw=0.0018, q_mvar=0.00058, name="load18" )
load19 = pp.create_load(net, b64, p_mw=0.0018, q_mvar=0.00058, name="load19" )
load20 = pp.create_load(net, b65, p_mw=0.0018, q_mvar=0.00058, name="load20" )
load21 = pp.create_load(net, b66, p_mw=0.0018, q_mvar=0.00058, name="load21" )
load22 = pp.create_load(net, b67, p_mw=0.0018, q_mvar=0.00058, name="load22" )
load23 = pp.create_load(net, b68, p_mw=0.0018, q_mvar=0.00058, name="load23" )
load24 = pp.create_load(net, b69, p_mw=0.0018, q_mvar=0.00058, name="load24" )

In [ ]:
pp.runpp(net)

In [47]:
net.res_bus

vm_pu  va_degree      p_mw    q_mvar
0   1.000000   0.000000  2.887723  1.567084
1   1.008110   0.774626  0.000000  0.000000
2   1.021332   0.816950 -3.000000 -1.000000
3   1.008170   0.752150  0.000000  0.000000
4   1.008211   0.744143  0.000000  0.000000
5   1.008170   0.752150  0.000000  0.000000
6   1.006640   0.610264  0.000000  0.000000
7   0.998882   0.544044  0.002200  0.000720
8   0.992078   0.485251  0.002200  0.000720
9   0.986234   0.434212  0.002200  0.000720
10  0.981355   0.391214  0.002200  0.000720
11  0.977446   0.356506  0.002200  0.000720
12  0.974512   0.330293  0.002200  0.000720
13  0.972554   0.312730  0.002200  0.000720
14  0.971574   0.303922  0.002200  0.000720
15  1.006682   0.602268  0.000000  0.000000
16  0.998924   0.536054  0.002200  0.000720
17  0.992120   0.477267  0.002200  0.000720
18  0.986276   0.426231  0.002200  0.000720
19  0.981398   0.383237  0.002200  0.000720
20  0.977489   0.348533  0.002200  0.000720
21  0.974555   0.322322  0.002200  0.000720
22  0.972597   0.304760  0.002200  0.000720
23  0.971618   0.295953  0.002200  0.000720
24  1.006640   0.610264  0.000000  0.000000
25  0.998882   0.544044  0.002200  0.000720
26  0.992078   0.485251  0.002200  0.000720
27  0.986234   0.434212  0.002200  0.000720
28  0.981355   0.391214  0.002200  0.000720
29  0.977446   0.356506  0.002200  0.000720
30  0.974512   0.330293  0.002200  0.000720
31  0.972554   0.312730  0.002200  0.000720
32  0.971574   0.303922  0.002200  0.000720

In [48]:
net.res_line


p_from_mw  q_from_mvar   p_to_mw  q_to_mvar     pl_mw   ql_mvar  \
0   -2.961942    -1.123698  3.000000   1.000000  0.038058 -0.123698   
1    0.019436    -0.159481 -0.019408  -0.006029  0.000029 -0.165510   
2    0.019454    -0.187069 -0.019408  -0.006029  0.000046 -0.193098   
3    0.019436    -0.159481 -0.019408  -0.006029  0.000029 -0.165510   
4    0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
5    0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
6    0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
7    0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
8    0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
9    0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
10   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
11   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   
12   0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
13   0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
14   0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
15   0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
16   0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
17   0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
18   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
19   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   
20   0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
21   0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
22   0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
23   0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
24   0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
25   0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
26   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
27   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   

    i_from_ka   i_to_ka      i_ka  vm_from_pu  va_from_degree  vm_to_pu  \
0    0.090715  0.089380  0.090715    1.008110        0.774626  1.021332   
1    0.004601  0.000582  0.004601    1.008110        0.774626  1.008170   
2    0.005386  0.000582  0.005386    1.008110        0.774626  1.008211   
3    0.004601  0.000582  0.004601    1.008110        0.774626  1.008170   
4    0.027226  0.027226  0.027226    1.006640        0.610264  0.998882   
5    0.023881  0.023881  0.023881    0.998882        0.544044  0.992078   
6    0.020513  0.020513  0.020513    0.992078        0.485251  0.986234   
7    0.017126  0.017126  0.017126    0.986234        0.434212  0.981355   
8    0.013721  0.013721  0.013721    0.981355        0.391214  0.977446   
9    0.010303  0.010303  0.010303    0.977446        0.356506  0.974512   
10   0.006874  0.006874  0.006874    0.974512        0.330293  0.972554   
11   0.003439  0.003439  0.003439    0.972554        0.312730  0.971574   
12   0.027225  0.027225  0.027225    1.006682        0.602268  0.998924   
13   0.023880  0.023880  0.023880    0.998924        0.536054  0.992120   
14   0.020513  0.020513  0.020513    0.992120        0.477267  0.986276   
15   0.017125  0.017125  0.017125    0.986276        0.426231  0.981398   
16   0.013720  0.013720  0.013720    0.981398        0.383237  0.977489   
17   0.010302  0.010302  0.010302    0.977489        0.348533  0.974555   
18   0.006874  0.006874  0.006874    0.974555        0.322322  0.972597   
19   0.003439  0.003439  0.003439    0.972597        0.304760  0.971618   
20   0.027226  0.027226  0.027226    1.006640        0.610264  0.998882   
21   0.023881  0.023881  0.023881    0.998882        0.544044  0.992078   
22   0.020513  0.020513  0.020513    0.992078        0.485251  0.986234   
23   0.017126  0.017126  0.017126    0.986234        0.434212  0.981355   
24   0.013721  0.013721  0.013721    0.981355        0.391214  0.9774

In [49]:
net.res_trafo

p_hv_mw  q_hv_mvar   p_lv_mw  q_lv_mvar     pl_mw   ql_mvar   i_hv_ka  \
0 -2.887723  -1.567084  2.903616   1.629728  0.015892  0.062644  0.017245   
1  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   
2  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   
3  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   

    i_lv_ka  vm_hv_pu  va_hv_degree  vm_lv_pu  va_lv_degree  loading_percent  
0  0.095347  1.000000      0.000000  1.008110      0.774626        13.211717  
1  0.027226  1.008170      0.752150  1.006640      0.610264         5.039442  
2  0.027225  1.008211      0.744143  1.006682      0.602268         5.039250  
3  0.027226  1.008170      0.752150  1.006640      0.610264         5.039442

3

A. There are no violations in voltage ,lines and transformers

B. B. Create and display a colormap of the buses showing blue for all voltages below 0.95, green for voltages between 0.95 and 1.05, and red for all voltages above 1.05 pu.

In [ ]:
import pandapower.plotting as plot
pp.runpp(net)
cmap_list = [((0, 0.95), "blue"), ((0.95, 1.05), "green"), ((1.05, 1.1), "red")]
cmap, norm = plot.cmap_discrete(cmap_list)
bc = plot.create_bus_collection(net, net.bus.index, size=0.1, zorder=2, cmap=cmap, norm=norm)
lc = create_line_collection(net)
plot.draw_collections([lc, bc])

 4.If we consider that each house will purchase an electric vehicle with a 7 kW charger (in 
addition to their normal 2 kW with PF=0.95 load) 

In [ ]:
#create load
load1 = pp.create_load(net, b42, p_mw=0.005, q_mvar=0.00162, name="load1" )
load2 = pp.create_load(net, b43, p_mw=0.005, q_mvar=0.00162, name="load2" )
load3 = pp.create_load(net, b44, p_mw=0.005, q_mvar=0.00162, name="load3" )
load4 = pp.create_load(net, b45, p_mw=0.005, q_mvar=0.00162, name="load4" )
load5 = pp.create_load(net, b46, p_mw=0.005, q_mvar=0.00162, name="load5" )
load6 = pp.create_load(net, b47, p_mw=0.005, q_mvar=0.00162, name="load6" )
load7 = pp.create_load(net, b48, p_mw=0.005, q_mvar=0.00162, name="load7" )
load8 = pp.create_load(net, b49, p_mw=0.005, q_mvar=0.00162, name="load8" )

load9 = pp.create_load(net, b52, p_mw=0.005, q_mvar=0.00162, name="load9" )
load10 = pp.create_load(net, b53, p_mw=0.005, q_mvar=0.00162, name="load10" )
load11 = pp.create_load(net, b54, p_mw=0.005, q_mvar=0.00162, name="load11" )
load12 = pp.create_load(net, b55, p_mw=0.005, q_mvar=0.00162, name="load12" )
load13 = pp.create_load(net, b56, p_mw=0.005, q_mvar=0.00162, name="load13" )
load14 = pp.create_load(net, b57, p_mw=0.005, q_mvar=0.00162, name="load14" )
load15 = pp.create_load(net, b58, p_mw=0.005, q_mvar=0.00162, name="load15" )
load16 = pp.create_load(net, b59, p_mw=0.005, q_mvar=0.00162, name="load16" )

load17 = pp.create_load(net, b62, p_mw=0.005, q_mvar=0.00162, name="load17" )
load18 = pp.create_load(net, b63, p_mw=0.005, q_mvar=0.00162, name="load18" )
load19 = pp.create_load(net, b64, p_mw=0.005, q_mvar=0.00162, name="load19" )
load20 = pp.create_load(net, b65, p_mw=0.005, q_mvar=0.00162, name="load20" )
load21 = pp.create_load(net, b66, p_mw=0.005, q_mvar=0.00162, name="load21" )
load22 = pp.create_load(net, b67, p_mw=0.005, q_mvar=0.00162, name="load22" )
load23 = pp.create_load(net, b68, p_mw=0.005, q_mvar=0.00162, name="load23" )
load24 = pp.create_load(net, b69, p_mw=0.005, q_mvar=0.00162, name="load24" )

In [ ]:
pp.runpp(net)

In [50]:
net.res_bus


vm_pu  va_degree      p_mw    q_mvar
0   1.000000   0.000000  2.887723  1.567084
1   1.008110   0.774626  0.000000  0.000000
2   1.021332   0.816950 -3.000000 -1.000000
3   1.008170   0.752150  0.000000  0.000000
4   1.008211   0.744143  0.000000  0.000000
5   1.008170   0.752150  0.000000  0.000000
6   1.006640   0.610264  0.000000  0.000000
7   0.998882   0.544044  0.002200  0.000720
8   0.992078   0.485251  0.002200  0.000720
9   0.986234   0.434212  0.002200  0.000720
10  0.981355   0.391214  0.002200  0.000720
11  0.977446   0.356506  0.002200  0.000720
12  0.974512   0.330293  0.002200  0.000720
13  0.972554   0.312730  0.002200  0.000720
14  0.971574   0.303922  0.002200  0.000720
15  1.006682   0.602268  0.000000  0.000000
16  0.998924   0.536054  0.002200  0.000720
17  0.992120   0.477267  0.002200  0.000720
18  0.986276   0.426231  0.002200  0.000720
19  0.981398   0.383237  0.002200  0.000720
20  0.977489   0.348533  0.002200  0.000720
21  0.974555   0.322322  0.002200  0.000720
22  0.972597   0.304760  0.002200  0.000720
23  0.971618   0.295953  0.002200  0.000720
24  1.006640   0.610264  0.000000  0.000000
25  0.998882   0.544044  0.002200  0.000720
26  0.992078   0.485251  0.002200  0.000720
27  0.986234   0.434212  0.002200  0.000720
28  0.981355   0.391214  0.002200  0.000720
29  0.977446   0.356506  0.002200  0.000720
30  0.974512   0.330293  0.002200  0.000720
31  0.972554   0.312730  0.002200  0.000720
32  0.971574   0.303922  0.002200  0.000720

In [51]:
net.res_line

p_from_mw  q_from_mvar   p_to_mw  q_to_mvar     pl_mw   ql_mvar  \
0   -2.961942    -1.123698  3.000000   1.000000  0.038058 -0.123698   
1    0.019436    -0.159481 -0.019408  -0.006029  0.000029 -0.165510   
2    0.019454    -0.187069 -0.019408  -0.006029  0.000046 -0.193098   
3    0.019436    -0.159481 -0.019408  -0.006029  0.000029 -0.165510   
4    0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
5    0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
6    0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
7    0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
8    0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
9    0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
10   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
11   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   
12   0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
13   0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
14   0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
15   0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
16   0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
17   0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
18   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
19   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   
20   0.018024     0.005974 -0.017892  -0.005907  0.000132  0.000067   
21   0.015692     0.005187 -0.015590  -0.005136  0.000102  0.000051   
22   0.013390     0.004416 -0.013315  -0.004378  0.000075  0.000038   
23   0.011115     0.003658 -0.011063  -0.003632  0.000052  0.000026   
24   0.008863     0.002912 -0.008829  -0.002895  0.000034  0.000017   
25   0.006629     0.002175 -0.006611  -0.002165  0.000019  0.000009   
26   0.004411     0.001445 -0.004402  -0.001441  0.000008  0.000004   
27   0.002202     0.000721 -0.002200  -0.000720  0.000002  0.000001   

    i_from_ka   i_to_ka      i_ka  vm_from_pu  va_from_degree  vm_to_pu  \
0    0.090715  0.089380  0.090715    1.008110        0.774626  1.021332   
1    0.004601  0.000582  0.004601    1.008110        0.774626  1.008170   
2    0.005386  0.000582  0.005386    1.008110        0.774626  1.008211   
3    0.004601  0.000582  0.004601    1.008110        0.774626  1.008170   
4    0.027226  0.027226  0.027226    1.006640        0.610264  0.998882   
5    0.023881  0.023881  0.023881    0.998882        0.544044  0.992078   
6    0.020513  0.020513  0.020513    0.992078        0.485251  0.986234   
7    0.017126  0.017126  0.017126    0.986234        0.434212  0.981355   
8    0.013721  0.013721  0.013721    0.981355        0.391214  0.977446   
9    0.010303  0.010303  0.010303    0.977446        0.356506  0.974512   
10   0.006874  0.006874  0.006874    0.974512        0.330293  0.972554   
11   0.003439  0.003439  0.003439    0.972554        0.312730  0.971574   
12   0.027225  0.027225  0.027225    1.006682        0.602268  0.998924   
13   0.023880  0.023880  0.023880    0.998924        0.536054  0.992120   
14   0.020513  0.020513  0.020513    0.992120        0.477267  0.986276   
15   0.017125  0.017125  0.017125    0.986276        0.426231  0.981398   
16   0.013720  0.013720  0.013720    0.981398        0.383237  0.977489   
17   0.010302  0.010302  0.010302    0.977489        0.348533  0.974555   
18   0.006874  0.006874  0.006874    0.974555        0.322322  0.972597   
19   0.003439  0.003439  0.003439    0.972597        0.304760  0.971618   
20   0.027226  0.027226  0.027226    1.006640        0.610264  0.998882   
21   0.023881  0.023881  0.023881    0.998882        0.544044  0.992078   
22   0.020513  0.020513  0.020513    0.992078        0.485251  0.986234   
23   0.017126  0.017126  0.017126    0.986234        0.434212  0.981355   
24   0.013721  0.013721  0.013721    0.981355        0.391214  0.9774

In [52]:
net.res_trafo

p_hv_mw  q_hv_mvar   p_lv_mw  q_lv_mvar     pl_mw   ql_mvar   i_hv_ka  \
0 -2.887723  -1.567084  2.903616   1.629728  0.015892  0.062644  0.017245   
1  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   
2  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   
3  0.019408   0.006029 -0.018024  -0.005974  0.001384  0.000056  0.000582   

    i_lv_ka  vm_hv_pu  va_hv_degree  vm_lv_pu  va_lv_degree  loading_percent  
0  0.095347  1.000000      0.000000  1.008110      0.774626        13.211717  
1  0.027226  1.008170      0.752150  1.006640      0.610264         5.039442  
2  0.027225  1.008211      0.744143  1.006682      0.602268         5.039250  
3  0.027226  1.008170      0.752150  1.006640      0.610264         5.039442

A.  This system is not able to withstand this load.
B. solution for problem may be we can use the another transformer to provide the voltage . 

# SHORT CIRCUIT ANALYSIS
1.Compute the maximum three-phase short-circuit currents for a solid fault (ZF=0 Ω).


A. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each bus.

In [53]:
import pandapower.shortcircuit as sc

sc.calc_sc(net, ip=True, ith=True, branch_results=True)
net.res_bus_sc

Branch results are in beta mode and might not always be reliable, especially for transformers


ikss_ka     skss_mw      ip_ka    ith_ka     rk_ohm      xk_ohm
0   0.544779  100.000000   1.039833  0.547460  43.969642  125.627547
1   2.154304   70.832867   4.435416  2.168940   1.517492    6.023599
2   1.835268   59.781143   3.243817  1.842178   3.082492    6.683599
3   1.757434   57.783918   3.051211  1.763578   3.395492    6.815599
4   1.699228   55.870123   2.899378  1.704790   3.708492    6.947599
5   1.757434   57.783918   3.051211  1.763578   3.395492    6.815599
6   9.835564    6.467809  19.915061  9.897797   0.007113    0.026265
7   3.072391    2.020386   4.544073  3.077455   0.066503    0.056265
8   1.753726    1.153240   2.556174  1.756279   0.125893    0.086265
9   1.223558    0.804605   1.776722  1.225272   0.185283    0.116265
10  0.938901    0.617416   1.361183  0.940193   0.244673    0.146265
11  0.761516    0.500768   1.103055  0.762553   0.304063    0.176265
12  0.640440    0.421149   0.927175  0.641307   0.363453    0.206265
13  0.552553    0.363356   0.799648  0.553298   0.422843    0.236265
14  0.485863    0.319501   0.702950  0.486516   0.482233    0.266265
15  9.805331    6.447927  19.781943  9.866438   0.007239    0.026318
16  3.067824    2.017383   4.536929  3.072877   0.066629    0.056318
17  1.752197    1.152235   2.553898  1.754747   0.126019    0.086318
18  1.222808    0.804112   1.775619  1.224521   0.185409    0.116318
19  0.938458    0.617125   1.360534  0.939749   0.244799    0.146318
20  0.761224    0.500576   1.102629  0.762261   0.304189    0.176318
21  0.640233    0.421013   0.926874  0.641100   0.363579    0.206318
22  0.552399    0.363254   0.799424  0.553144   0.422969    0.236318
23  0.485744    0.319422   0.702776  0.486397   0.482359    0.266318
24  9.835564    6.467809  19.915061  9.897797   0.007113    0.026265
25  3.072391    2.020386   4.544073  3.077455   0.066503    0.056265
26  1.753726    1.153240   2.556174  1.756279   0.125893    0.086265
27  1.223558    0.804605   1.776722  1.225272   0.185283    0.116265
28  0.938901    0.617416   1.361183  0.940193   0.244673    0.146265
29  0.761516    0.500768   1.103055  0.762553   0.304063    0.176265
30  0.640440    0.421149   0.927175  0.641307   0.363453    0.206265
31  0.552553    0.363356   0.799648  0.553298   0.422843    0.236265
32  0.485863    0.319501   0.702950  0.486516   0.482233    0.266265

B. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each line.

In [54]:
net.res_line_sc

ikss_ka     ip_ka    ith_ka
0   1.725733  3.088911  1.732230
1   1.757434  3.051211  1.763578
2   1.699228  2.899378  1.704790
3   1.757434  3.051211  1.763578
4   3.072391  4.544073  3.077455
5   1.753726  2.556174  1.756279
6   1.223558  1.776722  1.225272
7   0.938901  1.361183  0.940193
8   0.761516  1.103055  0.762553
9   0.640440  0.927175  0.641307
10  0.552553  0.799648  0.553298
11  0.485863  0.702950  0.486516
12  3.067824  4.536929  3.072877
13  1.752197  2.553898  1.754747
14  1.222808  1.775619  1.224521
15  0.938458  1.360534  0.939749
16  0.761224  1.102629  0.762261
17  0.640233  0.926874  0.641100
18  0.552399  0.799424  0.553144
19  0.485744  0.702776  0.486397
20  3.072391  4.544073  3.077455
21  1.753726  2.556174  1.756279
22  1.223558  1.776722  1.225272
23  0.938901  1.361183  0.940193
24  0.761516  1.103055  0.762553
25  0.640440  0.927175  0.641307
26  0.552553  0.799648  0.553298
27  0.485863  0.702950  0.486516

C. What is the difference between the three values of current computed at each 
node or line?

The values of all the three currents at the buses are little higher at some buses compare tp the lines .

for example , the values of currents are high ip= 19.9 ka ,ikss=9.80 ka, ith= 9.88 ka.
at lines the maximum current value is 4.5 ka.


D. Which standard is used to compute the short-circuit currents? Briefly explain 
the methodology based on the documentation, the slides of the course, and 
material you find online.

The standard used to compute the short circuit current is IEC 60909-0

Methods used for short circuit current calculations:

The measurement or calculation of short-circuit current in LV networks on final
circuits is very simple. In meshed and extensive power plants, the calculation is
more difficult because of the short-circuit current of several partial short-circuit
currents in conductors and earth return.

The short-circuit currents in three-phase systems can be determined by three
different calculation procedures:

1 superposition method for a defined load flow case
2 calculating with the equivalent voltage source at the fault location 
3 transient calculation
 
Superposition method:
     This is the exact method used for calcultaing short ciruit currents.It contains three steps
     First step , he currents, voltages, and internal voltages for steady-state operation before onset of the short circuit are      calculated.The calculation considers the impedances, power supply feeders,and node loads of the active elements. 
     In the second step, the voltage applied to the fault location before the occurrence of the short circuit and the current        distribution at the fault location are determined with a negative sign.
     In the third step, both the conditions are superimposed. We then obtain a zero voltage at the fault location. The              superposition of the currents also leads to the value zero.
     
Transient Calculation:
   With the transient method, the individual operating equipment and, as a result,the entire network are represented by a          system of differential equations. The calculation is very tedious. The method with the equivalent voltage source is a
   simplification relative to the other methods. Since 1988, it has been standardized internationally in IEC 60909-0. The          calculation is independent of a current operational state


2.Deactivate the contribution of the large motor4 and re-compute the maximum three-phase short-circuit currents for a solid fault (ZF=0 Ω).

A. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each bus.

In [ ]:
net.sgen.in_service=False

In [55]:
import pandapower.shortcircuit as sc

sc.calc_sc(net, ip=True, ith=True, branch_results=True)
net.res_bus_sc

Branch results are in beta mode and might not always be reliable, especially for transformers


ikss_ka     skss_mw      ip_ka    ith_ka     rk_ohm      xk_ohm
0   0.544779  100.000000   1.039833  0.547460  43.969642  125.627547
1   2.154304   70.832867   4.435416  2.168940   1.517492    6.023599
2   1.835268   59.781143   3.243817  1.842178   3.082492    6.683599
3   1.757434   57.783918   3.051211  1.763578   3.395492    6.815599
4   1.699228   55.870123   2.899378  1.704790   3.708492    6.947599
5   1.757434   57.783918   3.051211  1.763578   3.395492    6.815599
6   9.835564    6.467809  19.915061  9.897797   0.007113    0.026265
7   3.072391    2.020386   4.544073  3.077455   0.066503    0.056265
8   1.753726    1.153240   2.556174  1.756279   0.125893    0.086265
9   1.223558    0.804605   1.776722  1.225272   0.185283    0.116265
10  0.938901    0.617416   1.361183  0.940193   0.244673    0.146265
11  0.761516    0.500768   1.103055  0.762553   0.304063    0.176265
12  0.640440    0.421149   0.927175  0.641307   0.363453    0.206265
13  0.552553    0.363356   0.799648  0.553298   0.422843    0.236265
14  0.485863    0.319501   0.702950  0.486516   0.482233    0.266265
15  9.805331    6.447927  19.781943  9.866438   0.007239    0.026318
16  3.067824    2.017383   4.536929  3.072877   0.066629    0.056318
17  1.752197    1.152235   2.553898  1.754747   0.126019    0.086318
18  1.222808    0.804112   1.775619  1.224521   0.185409    0.116318
19  0.938458    0.617125   1.360534  0.939749   0.244799    0.146318
20  0.761224    0.500576   1.102629  0.762261   0.304189    0.176318
21  0.640233    0.421013   0.926874  0.641100   0.363579    0.206318
22  0.552399    0.363254   0.799424  0.553144   0.422969    0.236318
23  0.485744    0.319422   0.702776  0.486397   0.482359    0.266318
24  9.835564    6.467809  19.915061  9.897797   0.007113    0.026265
25  3.072391    2.020386   4.544073  3.077455   0.066503    0.056265
26  1.753726    1.153240   2.556174  1.756279   0.125893    0.086265
27  1.223558    0.804605   1.776722  1.225272   0.185283    0.116265
28  0.938901    0.617416   1.361183  0.940193   0.244673    0.146265
29  0.761516    0.500768   1.103055  0.762553   0.304063    0.176265
30  0.640440    0.421149   0.927175  0.641307   0.363453    0.206265
31  0.552553    0.363356   0.799648  0.553298   0.422843    0.236265
32  0.485863    0.319501   0.702950  0.486516   0.482233    0.266265

B. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each line.


In [56]:
net.res_line_sc

ikss_ka     ip_ka    ith_ka
0   1.725733  3.088911  1.732230
1   1.757434  3.051211  1.763578
2   1.699228  2.899378  1.704790
3   1.757434  3.051211  1.763578
4   3.072391  4.544073  3.077455
5   1.753726  2.556174  1.756279
6   1.223558  1.776722  1.225272
7   0.938901  1.361183  0.940193
8   0.761516  1.103055  0.762553
9   0.640440  0.927175  0.641307
10  0.552553  0.799648  0.553298
11  0.485863  0.702950  0.486516
12  3.067824  4.536929  3.072877
13  1.752197  2.553898  1.754747
14  1.222808  1.775619  1.224521
15  0.938458  1.360534  0.939749
16  0.761224  1.102629  0.762261
17  0.640233  0.926874  0.641100
18  0.552399  0.799424  0.553144
19  0.485744  0.702776  0.486397
20  3.072391  4.544073  3.077455
21  1.753726  2.556174  1.756279
22  1.223558  1.776722  1.225272
23  0.938901  1.361183  0.940193
24  0.761516  1.103055  0.762553
25  0.640440  0.927175  0.641307
26  0.552553  0.799648  0.553298
27  0.485863  0.702950  0.486516

C. Compare to the values of Task 1 and explain the difference.

 The values of the all the currents ramin same as the task 1 
 decativation of motor doesnot affect the short circuit currents at the line and buses.


3 Compute the maximum three-phase short-circuit currents with fault impedance 
ZF=1+j1.5 Ω.

A. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each bus.


In [ ]:
net.sgen.in_service=True

In [57]:
import pandapower.shortcircuit as sc

sc.calc_sc(net, ip=True, ith=True, branch_results=True, r_fault_ohm=1, x_fault_ohm=1.5)
net.res_bus_sc

Branch results are in beta mode and might not always be reliable, especially for transformers


ikss_ka    skss_mw     ip_ka    ith_ka     rk_ohm      xk_ohm
0   0.537980  98.704543  1.028554  0.540645  44.969642  127.127547
1   1.710533  55.460188  3.488893  1.721903   2.517492    7.523599
2   1.498400  48.111706  2.753461  1.505015   4.082492    8.183599
3   1.439765  46.779523  2.611587  1.445753   4.395492    8.315599
4   1.399752  45.495973  2.502729  1.405265   4.708492    8.447599
5   1.439765  46.779523  2.611587  1.445753   4.395492    8.315599
6   0.639009   0.096249  0.994130  0.643114   1.007113    1.526265
7   0.290864   0.093288  0.495668  0.292644   1.066503    1.556265
8   0.219762   0.090478  0.389523  0.221047   1.125893    1.586265
9   0.188956   0.087811  0.340833  0.190015   1.185283    1.616265
10  0.170827   0.085278  0.310493  0.171746   1.244673    1.646265
11  0.158333   0.082871  0.288505  0.159153   1.304063    1.676265
12  0.148873   0.080582  0.271153  0.149617   1.363453    1.706265
13  0.141260   0.078404  0.256727  0.141942   1.422843    1.736265
14  0.134875   0.076329  0.244322  0.135506   1.482233    1.766265
15  0.637463   0.096243  0.991931  0.641558   1.007239    1.526318
16  0.290624   0.093282  0.495316  0.292402   1.066629    1.556318
17  0.219676   0.090473  0.389391  0.220961   1.126019    1.586318
18  0.188911   0.087806  0.340758  0.189969   1.185409    1.616318
19  0.170797   0.085273  0.310442  0.171716   1.244799    1.646318
20  0.158312   0.082866  0.288466  0.159132   1.304189    1.676318
21  0.148856   0.080578  0.271121  0.149600   1.363579    1.706318
22  0.141246   0.078399  0.256700  0.141928   1.422969    1.736318
23  0.134864   0.076325  0.244298  0.135494   1.482359    1.766318
24  0.639009   0.096249  0.994130  0.643114   1.007113    1.526265
25  0.290864   0.093288  0.495668  0.292644   1.066503    1.556265
26  0.219762   0.090478  0.389523  0.221047   1.125893    1.586265
27  0.188956   0.087811  0.340833  0.190015   1.185283    1.616265
28  0.170827   0.085278  0.310493  0.171746   1.244673    1.646265
29  0.158333   0.082871  0.288505  0.159153   1.304063    1.676265
30  0.148873   0.080582  0.271153  0.149617   1.363453    1.706265
31  0.141260   0.078404  0.256727  0.141942   1.422843    1.736265
32  0.134875   0.076329  0.244322  0.135506   1.482233    1.766265

B. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each line.


In [58]:
net.res_line_sc

ikss_ka     ip_ka    ith_ka
0   1.388865  2.598556  1.394997
1   1.439765  2.611587  1.445753
2   1.399752  2.502729  1.405265
3   1.439765  2.611587  1.445753
4   0.290864  0.495668  0.292644
5   0.219762  0.389523  0.221047
6   0.188956  0.340833  0.190015
7   0.170827  0.310493  0.171746
8   0.158333  0.288505  0.159153
9   0.148873  0.271153  0.149617
10  0.141260  0.256727  0.141942
11  0.134875  0.244322  0.135506
12  0.290624  0.495316  0.292402
13  0.219676  0.389391  0.220961
14  0.188911  0.340758  0.189969
15  0.170797  0.310442  0.171716
16  0.158312  0.288466  0.159132
17  0.148856  0.271121  0.149600
18  0.141246  0.256700  0.141928
19  0.134864  0.244298  0.135494
20  0.290864  0.495668  0.292644
21  0.219762  0.389523  0.221047
22  0.188956  0.340833  0.190015
23  0.170827  0.310493  0.171746
24  0.158333  0.288505  0.159153
25  0.148873  0.271153  0.149617
26  0.141260  0.256727  0.141942
27  0.134875  0.244322  0.135506

C. Compare to the values of Task 1 and explain the difference.
   
   The values of the short circuit currents at buses and lines are reduced when compared to task 1 with fault impedance 0 ohm.
   the max value of short ciruit current when the fault impedance added is 3.5 ka at buses and 2.5ka at lines.

4.Compute the maximum two-phase short-circuit currents for a solid fault (ZF=0 Ω).

A. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each bus.

In [59]:
import pandapower.shortcircuit as sc

sc.calc_sc(net, fault="2ph", ip=True, ith=True, branch_results=True)
net.res_bus_sc

Branch results are in beta mode and might not always be reliable, especially for transformers


ikss_ka    skss_mw      ip_ka    ith_ka     rk_ohm      xk_ohm
0   0.474461  28.867513   0.904295  0.476796  43.969642  125.627547
1   1.880357  20.447687   3.861936  1.893132   1.517492    6.023599
2   1.604063  17.257330   2.829981  1.610103   3.082492    6.683599
3   1.533954  16.680780   2.659356  1.539317   3.395492    6.815599
4   1.483150  16.128315   2.527304  1.488004   3.708492    6.947599
5   1.533954  16.680780   2.659356  1.539317   3.395492    6.815599
6   8.584847   1.867096  17.341699  8.639166   0.007113    0.026265
7   2.681698   0.583235   3.964880  2.686118   0.066503    0.056265
8   1.530717   0.332912   2.230606  1.532945   0.125893    0.086265
9   1.067967   0.232269   1.550474  1.069463   0.185283    0.116265
10  0.819508   0.178233   1.187864  0.820636   0.244673    0.146265
11  0.664679   0.144559   0.962610  0.665585   0.304063    0.176265
12  0.559000   0.121575   0.809127  0.559756   0.363453    0.206265
13  0.482289   0.104892   0.697839  0.482939   0.422843    0.236265
14  0.424079   0.092232   0.613453  0.424649   0.482233    0.266265
15  8.558458   1.861356  17.226125  8.611795   0.007239    0.026318
16  2.677711   0.582368   3.958650  2.682122   0.066629    0.056318
17  1.529383   0.332622   2.228620  1.531609   0.126019    0.086318
18  1.067313   0.232127   1.549511  1.068807   0.185409    0.116318
19  0.819121   0.178149   1.187298  0.820248   0.244799    0.146318
20  0.664425   0.144504   0.962238  0.665330   0.304189    0.176318
21  0.558819   0.121536   0.808864  0.559576   0.363579    0.206318
22  0.482155   0.104863   0.697643  0.482805   0.422969    0.236318
23  0.423975   0.092209   0.613302  0.424545   0.482359    0.266318
24  8.584847   1.867096  17.341699  8.639166   0.007113    0.026265
25  2.681698   0.583235   3.964880  2.686118   0.066503    0.056265
26  1.530717   0.332912   2.230606  1.532945   0.125893    0.086265
27  1.067967   0.232269   1.550474  1.069463   0.185283    0.116265
28  0.819508   0.178233   1.187864  0.820636   0.244673    0.146265
29  0.664679   0.144559   0.962610  0.665585   0.304063    0.176265
30  0.559000   0.121575   0.809127  0.559756   0.363453    0.206265
31  0.482289   0.104892   0.697839  0.482939   0.422843    0.236265
32  0.424079   0.092232   0.613453  0.424649   0.482233    0.266265

B. Present the initial symmetrical short-circuit current, the short-circuit current 
peak, and the equivalent thermal short-circuit current at each line.


In [60]:
net.res_line_sc

ikss_ka     ip_ka    ith_ka
0   1.494529  2.675075  1.500155
1   1.533954  2.659356  1.539317
2   1.483150  2.527304  1.488004
3   1.533954  2.659356  1.539317
4   2.681698  3.964880  2.686118
5   1.530717  2.230606  1.532945
6   1.067967  1.550474  1.069463
7   0.819508  1.187864  0.820636
8   0.664679  0.962610  0.665585
9   0.559000  0.809127  0.559756
10  0.482289  0.697839  0.482939
11  0.424079  0.613453  0.424649
12  2.677711  3.958650  2.682122
13  1.529383  2.228620  1.531609
14  1.067313  1.549511  1.068807
15  0.819121  1.187298  0.820248
16  0.664425  0.962238  0.665330
17  0.558819  0.808864  0.559576
18  0.482155  0.697643  0.482805
19  0.423975  0.613302  0.424545
20  2.681698  3.964880  2.686118
21  1.530717  2.230606  1.532945
22  1.067967  1.550474  1.069463
23  0.819508  1.187864  0.820636
24  0.664679  0.962610  0.665585
25  0.559000  0.809127  0.559756
26  0.482289  0.697839  0.482939
27  0.424079  0.613453  0.424649

C. Compare to the values of Task 1 and explain the difference.

 In 2 phase short circuit current with solid fault , the values of current is slightly reduced compare to the task with 3 phase short circuit.
 
 
